plugins/out_s3/s3.c

```diff
@@ -1385,8 +1411,8 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
         c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
                                               uri, final_body, final_body_size,
                                               headers, num_headers);
-        if (ctx->compression != FLB_AWS_COMPRESS_NONE) {
-             flb_free(compressed_body);
+        if (ctx->compression != FLB_AWS_COMPRESS_NONE && ctx->is_compressed == FLB_TRUE) {
+            flb_free(compressed_body);
```

Comment:
Why modify the code in put_object? I don't undertand why this code needs to be modified to support multipart compression... why change the logic here?

Reply:
So we move compress data ahead. how can we know the current data is compressed or not